### ISSUE TITLE
TacticianMiddlewareBundle

### ISSUE THREAD

**Issue by xtrasmal** [xtrasmal/TacticianBundle#20]
Will middleware be added to tactician library or will there be a separate library for that?
Personally I don't mind adding a separate bundle with loads of middleware to any of my projects. 

https://github.com/xtrasmal/TacticianMiddlewareBundle

**Comment by rdohms** [comment]
My 2 cents.

My preferences would be:
- If the middleware is broad, have it under the library in its own package (like bernard) and then have the bundle have options to wire it in.
- if its symfony specific, have it in the bundle and wired to be dropped in

The disadvantage to the MiddlewareBundle is that composer can't deal with those dependencies and the Bundle would simply be a dump of middlewares with little extra purpose.

**Comment by rtuin** [comment]
I think the validatormiddleware could be extracted in a separate package. It doesn't even need to be a bundle since apart from tactician it only depends on a symfony component, and not the framework bundle for example.

By doing this you will lose the service definition for the validatormiddleware, but gain the possibility of using the middleware outside  symfony framework projects.

**Comment by rdohms** [comment]
The decision to keep it in the bundle is the excessive number and fragmentation of tactician middlewares out there and poor @rosstuck keeping track of it all.

Even if it is separate package, I would like the service definition to remain so that this bundle has less boilerplate and more drop-in action going on. Especially since it has no effect unless listed in middleware list. No need to force people to do even more by defining the service first.

**Comment by rtuin** [comment]
Okay I agree we need to find a better solution to that. But a bundle is not the best place then.

I didn't realise you can define services that point to none-existing classes, even if you're not using them. that's useful indeed!

**Comment by xtrasmal** [comment]
The idea of a drop-in mega package of cool decorators and/or middleware was the initial idea of starting this bundle. @rosstuck spoke about having loads of middleware/decorators one day if I recall that correctly. 

It would be nice if we could have a companion bundle or whatever that is optional and compatible.

**Comment by rosstuck** [comment]
@xtrasmal If we've got middleware and other add-ons that are specific to Symfony2 framework, then let's just put them here. There's no need for a separate bundle just for middleware. :) I'd vote let's wipe out the Middleware bundle and just focus efforts here.

@rtuin Originally, @rdohms was going to split out the Validator middleware but I was a little hesitant. While the Validator is technically a separate component, I've only ever seen one project that used it standalone (and all I heard about it was a lotta profanity). In the case of Doctrine, that is used standalone very often so it got its own Tactician package (and I'm planning to add it to this project as a dep)

We could do the same with Validation, I just don't know if there's actual demand for that enough to justify yet _another_ plugin package that has to be maintained (think Travis, deps management, release tagging, README, changelog, etc). If there is that demand or we'd like to share it with another framework using the validator (Silex? Laravel? Not sure if anyone is), we can easily split the Validator middleware out in the future and keep the service definition here as @rdohms suggests.

In short, we can split it out and I'm open to doing so, I just need a reason. :smiley_cat:
